### PR TITLE
Add key encryption support to nvmeof configuration

### PR DESCRIPTION
**Pull request by gbregman**
mgr/cephadm: Add key encryption support to nvmeof configuration
python-common/ceph/deployment: Add key encryption support to nvmeof configuration

    Fixes https://tracker.ceph.com/issues/69036

    Signed-off-by: Gil Bregman <gbregman@il.ibm.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by gbregman** [comment]
@VallariAg please include in your tests

File: src/python-common/ceph/deployment/service_spec.py [hunk 1]
```diff
@@ -1639,25 +1642,25 @@ def validate(self) -> None:
         if (self.max_namespaces_with_netmask and self.max_namespaces_with_netmask < 0):
             raise SpecValidationError("Max namespaces with netmask can't be negative")
 
-        if type(self.max_subsystems) != int:
+        if not isinstance(self.max_subsystems, int):
```
Comment (idryomov) [inline src/python-common/ceph/deployment/service_spec.py:1645]: I'd suggest that you split this and three similar fixups below into a separate PR to expedite addressing `make check` failures.

**Comment by VallariAg** [comment]
Build started: https://shaman.ceph.com/builds/ceph/wip-vallari2-2024-11-25-1535/

**Review by idryomov — APPROVED** [review]
The change of `max_hosts_per_namespace` default from 1 to 8 is unrelated to encryption and not explained in the commit messages or the tracker ticket, but I'm approving because this PR happens to address `make check` failures reported in https://github.com/ceph/ceph/pull/60769#discussion_r1855543545 (also unrelated to encryption).

**Comment by VallariAg** [comment]
"nvmeof" teuthology suite passed for this PR: https://pulpito.ceph.com/vallariag-2024-11-25_12:07:35-nvmeof-wip-vallari2-2024-11-25-1535-distro-default-smithi/

**Pull request merged**